frontend/src/buttons/CSVDownloadButton/CSVDownloadButton.tsx

```diff
@@ -0,0 +1,38 @@
+import { IconButton, Tooltip } from '@material-ui/core'
+import React, { useEffect, useState } from 'react'
+import { Icon } from '../../components/Icon'
+import { Dispatch, ApplicationState } from '../../store'
+import { useDispatch, useSelector } from 'react-redux'
+
+interface Props {
+  deviceID: string
+  maxDate: string
+}
+
+export function CSVDownloadButton({ deviceID, maxDate }: Props) {
+  const dispatch = useDispatch<Dispatch>()
+  const { getEventsURL, set } = dispatch.logs
+  const { eventsURL } = useSelector((state: ApplicationState) => state.logs)
+  const [shouldDownload, setShouldDownload] = useState(false)
+
+  useEffect(() => {
+    if (shouldDownload && eventsURL) {
+      window.open(eventsURL)
+      setShouldDownload(false)
+      set({ eventsURL: '' })
+    }
+  }, [eventsURL, shouldDownload])
```

Comment:
Just curious why you do this as a saved state and effect? Is that to prevent multiple clicks?

Reply:
we need to wait for the new eventURL after click in download button